src/components/TeacherComponents/helpers/downloadAndShareHelpers/fetchHubspotContactDetails.ts

```diff
@@ -21,16 +24,24 @@ export async function fetchHubspotContactDetails(email: string) {
       throw new Error("Failed to fetch contact details");
     }
 
+    if (response.status === 204) {
```

Comment:
This was the root. Our endpoint was already set up to handle missing contacts by responding with a 204, but we weren't handling it at all.